vm/rust/src/lib.rs

```diff
@@ -240,55 +276,78 @@ pub extern "C" fn cairoVMExecute(
 
         let mut txn_state = CachedState::create_transactional(&mut state);
         let fee_type;
+        let minimal_l1_gas_amount_vector: Option<GasVector>;
         let res = match txn.unwrap() {
             Transaction::AccountTransaction(t) => {
                 fee_type = t.fee_type();
+                minimal_l1_gas_amount_vector =
+                    Some(gas_usage::estimate_minimal_gas_vector(&block_context, &t).unwrap());
                 t.execute(&mut txn_state, &block_context, charge_fee, validate)
             }
             Transaction::L1HandlerTransaction(t) => {
                 fee_type = t.fee_type();
+                minimal_l1_gas_amount_vector = None;
                 t.execute(&mut txn_state, &block_context, charge_fee, validate)
             }
         };
 
         match res {
             Err(error) => {
                 let err_string = match &error {
-                    ContractConstructorExecutionFailed(e)
-                        | ExecutionError(e)
-                        | ValidateTransactionError(e) => format!("{error} {e}"),
-                    other => other.to_string()
+                    ContractConstructorExecutionFailed(e) => format!("{error} {e}"),
+                    ExecutionError { error: e, .. } | ValidateTransactionError { error: e, .. } => {
+                        format!("{error} {e}")
+                    }
+                    other => other.to_string(),
                 };
                 report_error(
                     reader_handle,
                     format!(
                         "failed txn {} reason: {}",
-                        txn_and_query_bit.txn_hash,
-                        err_string,
+                        txn_and_query_bit.txn_hash, err_string,
                     )
                     .as_str(),
-                    txn_index as i64
+                    txn_index as i64,
                 );
                 return;
             }
             Ok(mut t) => {
                 if t.is_reverted() && err_on_revert != 0 {
                     report_error(
                         reader_handle,
-                        format!("reverted: {}", t.revert_error.unwrap())
-                        .as_str(),
-                        txn_index as i64
+                        format!("reverted: {}", t.revert_error.unwrap()).as_str(),
+                        txn_index as i64,
                     );
                     return;
                 }
 
                 // we are estimating fee, override actual fee calculation
-                if  t.actual_fee.0 == 0 {
-                    t.actual_fee = calculate_tx_fee(&t.actual_resources, &block_context, &fee_type).unwrap();
+                if t.transaction_receipt.fee.0 == 0 {
```

Comment:
I guess calculate_tx_fee was removed from blockifier? (just wondering why we do the calculation manually here)

Reply:
> I guess calculate_tx_fee was removed from blockifier?

correct

> just wondering why we do the calculation manually here

Oh, I'll ping you in slack with message that contains all necessary context